### PR TITLE
feature/f9-teams (modified pull request from F9)

### PR DESCRIPTION
**Pull request by gordyt**
OK, this is the pull request that was submitted by F9 with the following modifications:

First, I rebased out this commit:

    8b52117 @Ignore failing or flakey tests so that a continuous build can be setup

Second, did a fresh rebase onto the `develop` branch.  The problem area was in `MailboxTestUtil.java`. Both F9 and the platform team had updated the `initProvisioning` method to correct some test issues.  I combined the changes in the manner I judged to be best, retaining what the platform team had done, but adding in some additional code changes from F9.

After that I ran the unit tests in `zm-mailbox/store` and got a clean test run.

@rupalid The unit tests that had been ignored in the commit that I rebased out are documented [here](https://github.com/Zimbra/zm-mailbox/pull/219).

### CODE REVIEW

**Comment by gordyt** [comment]
I noticed the Travis CI build failed.  Is that supposed to be running successfully now?

File: store/src/java-test/com/zimbra/cs/account/MockProvisioning.java [hunk 1]
```diff
@@ -56,6 +57,9 @@
  * @author ysasaki
  */
 public final class MockProvisioning extends Provisioning {
+    static {
+      LC.zimbra_attrs_directory.setDefault(System.getProperty("user.dir") + "/conf/attrs");
```
Comment (rohan-ambasta) [inline store/src/java-test/com/zimbra/cs/account/MockProvisioning.java]: Here, we are assuming that the user.dir will always be .../zm-mailbox/store
But that may not be the case for tests running from common or soap folder or from any zimbra extension like EWS
I think, System.getProperty("server.dir") should be used as we are doing in MailboxTestUtil.getZimbraServerDir(String zimbraServerDir)

File: store/src/java-test/com/zimbra/cs/mailbox/MailboxTestUtil.java [hunk 2]
```diff
@@ -85,21 +85,22 @@ public static void initProvisioning() throws Exception {
...
-    Provisioning.setInstance(new MockProvisioning());
+        zimbraServerDir = getZimbraServerDir(zimbraServerDir);
+        System.setProperty("log4j.configuration", "log4j-test.properties");
+        System.setProperty("zimbra.config", zimbraServerDir + "src/java-test/localconfig-test.xml");
+        LC.reload();
+
+        String timezonefilePath = zimbraServerDir + "src/java-test/timezones-test.ics";
+        File d = new File(timezonefilePath);
+        if (!d.exists()) {
+            throw new FileNotFoundException("zm-timezones repository not found. Please clone this repo in same directory as zm-mailbox before running this test.");
```
Comment (rohan-ambasta) [inline store/src/java-test/com/zimbra/cs/mailbox/MailboxTestUtil.java]: I think this has happened while merging.
We no longer refer to the timezones.ics file in zm-timezones repo, so we should use the following exception message.
```throw new FileNotFoundException("timezones.ics not found.");```

**Comment by gordyt** [comment]
@rohan-ambasta Thank you for the feedback.  I'll update the PR and incorporate your suggestions.

File: store/src/java-test/com/zimbra/cs/account/MockProvisioning.java (hunk 1 above)
Comment (gordyt) [inline store/src/java-test/com/zimbra/cs/account/MockProvisioning.java]: @rohan-ambasta I'm wondering if we even need this here at all.  Isn't the common pattern to call `MailboxTestUtil.initProvisioning()` first?  If so, that is already initializing `LC.zimbra_attrs_directory` correctly (using the method you described).

Should I just rebase out this change completely?

File: store/src/java-test/com/zimbra/cs/redolog/op/CreateMessageTest.java [hunk 3]
```diff
@@ -12,17 +12,12 @@
 import com.zimbra.cs.mime.ParsedMessageDataSource;
 import com.zimbra.cs.redolog.RedoLogInput;
 import com.zimbra.cs.redolog.RedoLogOutput;
+import org.junit.*;
```
Comment (fatkudu) [inline store/src/java-test/com/zimbra/cs/redolog/op/CreateMessageTest.java]: Not a fan of wildcard imports.

File: store/src/java/com/zimbra/cs/redolog/logger/FileLogWriter.java [hunk 4]
```diff
@@ -23,6 +23,16 @@
  */
 package com.zimbra.cs.redolog.logger;
 
+import com.zimbra.common.localconfig.DebugConfig;
+import com.zimbra.common.service.ServiceException;
+import com.zimbra.common.util.Constants;
+import com.zimbra.common.util.ZimbraLog;
+import com.zimbra.cs.account.Provisioning;
+import com.zimbra.cs.redolog.*;
```
Comment (fatkudu) [inline store/src/java/com/zimbra/cs/redolog/logger/FileLogWriter.java]: wildcard import - yuck.

File: store/src/java-test/com/zimbra/cs/account/MockProvisioning.java (hunk 1 above)
Comment (rohan-ambasta) [inline store/src/java-test/com/zimbra/cs/account/MockProvisioning.java]: @gordyt  Yes, we actually don't need this.

File: store/src/java-test/com/zimbra/cs/redolog/op/CreateMessageTest.java (hunk 3 above)
Comment (gordyt) [inline store/src/java-test/com/zimbra/cs/redolog/op/CreateMessageTest.java]: Gren I went ahead and updated this.  Combined the change with a fixup commit back into the original commit.

File: store/src/java/com/zimbra/cs/redolog/logger/FileLogWriter.java (hunk 4 above)
Comment (gordyt) [inline store/src/java/com/zimbra/cs/redolog/logger/FileLogWriter.java]: Ditto for this one.

File: store/src/java-test/com/zimbra/cs/account/MockProvisioning.java (hunk 1 above)
Comment (gordyt) [inline store/src/java-test/com/zimbra/cs/account/MockProvisioning.java]: @rohan-ambasta I rebased out this change in a fixup commit.

**Pull request merged**